### PR TITLE
plugins/kubernetes: Add env var to never add a preStop

### PR DESCRIPTION
**Pull request by rata**
* Add env variable to disable preStop injection in kubernetes plugin

Relates to #3080

The reason is the one explained in the bug: it's TOO much to modify all yamls to add this annotation. Just having an option to change the default is way more easy for us.  What do you think? Does it make sense? 

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low

### CODE REVIEW

File: plugins/kubernetes/README.md [hunk 1]
```diff
@@ -185,6 +185,8 @@ Samson automatically adds `container[].lifecycle.preStop` `sleep 3` if a preStop
 `container[].samson/preStop` is not set to `disabled`, to prevent in-flight requests from getting lost when taking a pod
 out of rotation (alternatively set `metadata.annoations.container-nameofcontainer-samson/preStop: disabled`).
 
+The default behavior can be to never add this by setting the `KUBERNETES_DONT_ADD_PRESTOP=false`
```
Comment (grosser) [inline plugins/kubernetes/README.md]: better to let people opt-in since this is kinda hacky anyway so only do it if`KUBERNETES_ADD_PRESTOP=true`

Comment (rata) [reply plugins/kubernetes/README.md]: Oh, will do it that way. Thanks!

Comment (rata) [reply plugins/kubernetes/README.md]: I didn't want to force changes on your side =)

**Comment by rata** [comment]
@grosser what about now? :)

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ class TemplateFiller
 
     CUSTOM_UNIQUE_LABEL_KEY = 'rc_unique_identifier'
     SECRET_PULLER_IMAGE = ENV['SECRET_PULLER_IMAGE'].presence
+    KUBERNETES_ADD_PRESTOP = !!(ENV.fetch('KUBERNETES_ADD_PRESTOP', false))
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]:
```suggestion
    KUBERNETES_ADD_PRESTOP = ENV['KUBERNETES_ADD_PRESTOP'] == 'true'
```

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 3]
```diff
@@ -509,6 +510,7 @@ def set_image_pull_secrets
     end
 
     def set_pre_stop
+      return if ! KUBERNETES_ADD_PRESTOP
```
Comment (grosser) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]:
```suggestion
      return unless KUBERNETES_ADD_PRESTOP
```

**Comment by rata** [comment]
Oh, this 4 lines make the test coverage go under the threshold. Will try to add some test tomorrow :)

**Comment by rata** [comment]
@grosser not sure how to add a test for that (sorry, I really should learn something about ruby sometime). Can you give me a hint? :)

**Comment by grosser** [comment]
looks pretty close, I'll fix it up :)

**Comment by rata** [comment]
@grosser thanks a lot! I'll take a look afterwards, I'm curious :)

Thanks again

**Pull request closed without merging**